### PR TITLE
Interpreter integration.

### PR DESCRIPTION
**Pull request by xeno-by**

### CODE REVIEW

**Comment by xeno-by** [comment]
Supersedes https://github.com/scalameta/scalahost/pull/78

**Comment by xeno-by** [comment]
/cc @vjovanov 

I took the liberty to replace macroUnit-based lookup with implDdef myself. The rest is just a rebase of your previous pull request onto the updated scalahost.

**Comment by xeno-by** [comment]
@vjovanov I'll be merging. Sorry for not waiting for a review, but this pull request is on critical path of something that I have to complete today. I'm, of course, not a fan of ninja-merging stuff that should be reviewed, but these are exceptional times, and I apologize for that.

**Comment by vjovanov** [comment]
Sure, I completely understand.

File: tests/src/test/scala/macros/SerializerMetaJoint.scala [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+trait SerializeMetaJointSuite extends FunSuite {
+  // NOTE: In memory of: https://github.com/scalareflect/core/commit/7f3058d1293b46fc255f525a92deaeec4d64026d
+  // Initial commit
+  // master
+  //  scalareflect  authored on Mar 8, 2014
+  test("San Francisco 2015 (joint compilation)") {
+    val list: List = Cons(1, Cons(2, Nil))
+    assert(serialize(list) === """{"head": 1, "tail": {"head": 2, "tail": {$tag: 1}, $tag: 0}, $tag: 0}""")
+  }
+}
```
Comment (vjovanov) [inline tests/src/test/scala/macros/SerializerMetaJoint.scala]: You can get rid of these by setting
```
   "ensure_newline_at_eof_on_save": true	
```
in sublime user/default preferences.

Comment (xeno-by) [inline tests/src/test/scala/macros/SerializerMetaJoint.scala]: Done

**Pull request merged**